### DISCUSSION TITLE
ModuleNotFoundError: No module named 'pyranges.pyranges' when load pkl file

### DISCUSSION THREAD

**Discussion by jding5066** [aertslab/scenicplus#307] [Q&A]
Hi,

I updated my scenic+ environment to the lastest version '1.0.1.dev4+ge4bdd9f' recently and have been getting this error when trying to load scenicplus object that was created under an older version of scenic+.

Unfortunately I have deleted the old senicplus environment and am having trouble opening this file. Any ideas what I could try to solve this issue?

Thank you,
Jane

`import dill 
scplus_obj = dill.load(open(os.path.join(work_dir, 'scenicplus/scplus_obj.pkl'), 'rb'))`

`---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[93], line 2
      1 import dill 
----> 2 scplus_obj = dill.load(open(os.path.join(work_dir, 'scenicplus/scplus_obj.pkl'), 'rb'))

File ~/miniconda3/envs/Scenicplus/lib/python3.8/site-packages/dill/_dill.py:270, in load(file, ignore, **kwds)
    268 def load(file, ignore=None, **kwds):
    269     """unpickle an object from a file"""
--> 270     return Unpickler(file, ignore=ignore, **kwds).load()

File ~/miniconda3/envs/Scenicplus/lib/python3.8/site-packages/dill/_dill.py:473, in Unpickler.load(self)
    472 def load(self): #NOTE: if settings change, need to update attributes
--> 473     obj = StockUnpickler.load(self)
    474     if type(obj).__module__ == getattr(_main_module, '__name__', '__main__'):
    475         if not self._ignore:
    476             # point obj class to main

File ~/miniconda3/envs/Scenicplus/lib/python3.8/site-packages/dill/_dill.py:463, in Unpickler.find_class(self, module, name)
    461     return type(None) #XXX: special case: NoneType missing
    462 if module == 'dill.dill': module = 'dill._dill'
--> 463 return StockUnpickler.find_class(self, module, name)

ModuleNotFoundError: No module named 'pyranges.pyranges'
`

**Comment by SeppeDeWinter** [comment]
Hi @jding5066 

What version of pyranges have you installed?

Can you try again with pyranges version 0.0.111.

All the best,

Seppe

**Reply by jding5066** [reply]
Hi Seppe,

I tried using pyranges 0.0.111 but this time it gives me another error-

```
>>> import pyranges
>>> print(pyranges.__version__)
0.0.111
>>> scplus_obj = dill.load(open(os.path.join(work_dir, 'scenicplus-screen/scplus_obj.pkl'), 'rb'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "~/miniconda3/envs/Scenicplus/lib/python3.8/site-packages/dill/_dill.py", line 287, in load
    return Unpickler(file, ignore=ignore, **kwds).load()
  File "~/miniconda3/envs/Scenicplus/lib/python3.8/site-packages/dill/_dill.py", line 442, in load
    obj = StockUnpickler.load(self)
  File "~/miniconda3/envs/Scenicplus/lib/python3.8/site-packages/dill/_dill.py", line 432, in find_class
    return StockUnpickler.find_class(self, module, name)
ModuleNotFoundError: No module named 'pyranges.pyranges_main'
```

Do you know if there's other versions of pyranges that I could possibly try?

Thank you!

**Reply by SeppeDeWinter** [reply]
Hi @jding5066 

The latest version of pyranges seems to have this `pyranges_main` file, could you try with that one (version 0.0.129).

All the best,

Seppe

**Reply by jding5066** [reply]
Hi Seppe,

I installed a previous version of scenicplus (https://github.com/aertslab/scenicplus.git@a57717f969f34d48b9c5f288a1d71ef15bc1436f) and pyranges==0.0.111 and it solved the probem! 

Thank you for your help!

**Reply by SeppeDeWinter** [reply]
Great!

Good luck with your analyses.

All the best,

Seppe
